spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/FunctionConfiguration.java

```diff
@@ -426,37 +535,35 @@ public void afterPropertiesSet() throws Exception {
 					&& ObjectUtils.isEmpty(applicationContext.getBeanNamesForAnnotation(EnableBinding.class))
 					&& this.determineFunctionName(functionCatalog, environment)) {
 				BeanDefinitionRegistry registry = (BeanDefinitionRegistry) applicationContext.getBeanFactory();
-				RootBeanDefinition rootBeanDefinition = new RootBeanDefinition(BindableFunctionProxyFactory.class);
-				FunctionInvocationWrapper function = functionCatalog.lookup(streamFunctionProperties.getDefinition());
-				if (function != null) {
-					if (function.isSupplier()) {
-						this.inputCount = 0;
-						this.outputCount = 1;
-					}
-					else if (function.isConsumer()) {
-						this.inputCount = 1;
-						this.outputCount = 0;
-					}
-					else {
-						this.inputCount = 1;
-						this.outputCount = 1;
+				String[] functionDefinitions = streamFunctionProperties.getDefinition().split(";");
+				boolean nameBasedOnFunctionName = functionDefinitions.length > 1;
+				for (String functionDefinition : functionDefinitions) {
+					RootBeanDefinition functionBindableProxyDefinition = new RootBeanDefinition(BindableFunctionProxyFactory.class);
```

Comment:
Consider to use `BeanDefinitionBuilder` instead, so you wouldn't need to call `getConstructorArgumentValues()` all the time.